managed/devops/opscli/ybops/cloud/aws/cloud.py

```diff
@@ -387,5 +388,16 @@ def start_instance(self, args):
             instance = ec2.Instance(id=args["id"])
             instance.start()
             instance.wait_until_running()
+            # The OS boot up may take some time,
+            # so retry untill the instance allows SSH connection to it via port 22.
+            retry_count = 0
+            sock = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+            while retry_count < 25:
+                time.sleep(5)
+                retry_count = retry_count + 1
+                result = sock.connect_ex((args["private_ip"], 22))
```

Comment:
What if ssh port is not 22? It could be customised inside the provider.

Reply:
yeah, missed it, but need to check how I can get that info here.

Reply:
added